repose-aggregator/extensions/simple-rbac/src/main/scala/org/openrepose/filters/simplerbac/SimpleRbacFilter.scala

```diff
@@ -0,0 +1,261 @@
+          ))
+        case _ =>
+          None
+      }
+    )
+    rbacWadl match {
+      case Some(r) =>
+        Option(configuration.getWadlOutput: String) match {
+          case Some(wadlOutput) =>
+            val wadlPath: Path = Paths.get(StringUriUtilities.formatUri(getPath(wadlOutput, configurationRoot))).toAbsolutePath
+            try {
+              if (Files.exists(wadlPath) && Files.isWritable(wadlPath) || !Files.exists(wadlPath) && Files.createFile(wadlPath) != null) {
+                Files.write(wadlPath, r.getBytes)
+                logger.debug("Saved WADL to: " + wadlPath)
+              } else {
+                logger.warn("Cannot write to WADL file: " + wadlPath)
+                logger.debug(s"Generated WADL:\n\n$r\n")
+              }
+            } catch {
+              case ex: IOException =>
+                logger.warn("Cannot write to WADL file: " + wadlPath, ex)
+                logger.debug(s"Generated WADL:\n\n$r\n")
+            }
+          case _ =>
+            logger.debug(s"Generated WADL:\n\n$r\n")
+        }
+        val uuid = UUID.randomUUID
+        validator = Validator.apply(
+          s"SimpleRbacValidator_$uuid",
+          new StreamSource(new ByteArrayInputStream(r.getBytes), "file://simple-rbac.wadl"),
+          config
+        )
+        initialized = true
+      case _ =>
+        logger.warn("Unable to generate the WADL; check the provided resources.")
+    }
+  }
+
+  override def isInitialized: Boolean = initialized
+
+  private def getHandlers: DispatchHandler = {
+    val handlers: util.List[ResultHandler] = new util.ArrayList[ResultHandler]
+    Option(configuration.getDelegating) match {
+      case Some(d) =>
+        handlers.add(new MethodLabelHandler)
+        handlers.add(new DelegationHandler(d.getQuality))
+      case _ =>
+        handlers.add(new ServletResultHandler)
+    }
+    handlers.add(new InstrumentedHandler)
+    handlers.add(new ApiCoverageHandler)
+    Option(configuration.getDotOutput) match {
+      case Some(d) =>
+        if (StringUtils.isNotBlank(d)) {
+          val dotPath: String = StringUriUtilities.formatUri(getPath(d, configurationRoot))
+          val out: File = new File(dotPath)
+          try {
+            if (out.exists && out.canWrite || !out.exists && out.createNewFile) {
+              handlers.add(new SaveDotHandler(out, true, true))
+            } else {
+              logger.warn("Cannot write to DOT file: " + dotPath)
+            }
+          } catch {
+            case ex: IOException =>
+              logger.warn("Cannot write to DOT file: " + dotPath, ex)
+          }
+        }
+      case _ =>
+    }
+    new DispatchHandler(handlers.toList:_*)
+  }
+
+  private def getPath(path: String, configRoot: String): String = {
+    val file: File = new File(path)
+    if (file.isAbsolute) {
+      file.getAbsolutePath
+    } else {
+      new File(configRoot, path).getAbsolutePath
```

Comment:
yeah, in a uri maybe a file, if you provide a root and a path it'll behave smart. if it's relative it appends to the root , if it's absolute it'll give you the absolute without the root.